lib/internal/http2/core.js

```
@@ -983,22 +986,79 @@ function emitClose(self, error) {
   self.emit('close');
 }
 
-function finishSessionDestroy(session, error) {
+function cleanupSession(session) {
   const socket = session[kSocket];
-  if (!socket.destroyed)
-    socket.destroy(error);
-
   session[kProxySocket] = undefined;
   session[kSocket] = undefined;
   session[kHandle] = undefined;
   session[kNativeFields] = new Uint8Array(kSessionUint8FieldCount);
-  socket[kSession] = undefined;
-  socket[kServer] = undefined;
+  if (socket) {
+    socket[kSession] = undefined;
+    socket[kServer] = undefined;
+  }
+}
+
+function finishSessionDestroy(session, error) {
+  debugSessionObj(session, 'finishSessionDestroy');
+
+  const handle = session[kHandle];
+  const socket = session[kSocket];
+  if (handle) handle.ondone = null;
+  cleanupSession(session);
+
+  if (socket && !socket.destroyed)
```

I don't think the destroyed check is necessary here

---

Also, the error would be potentially swallowed here